### PR TITLE
Add error boundary at the app level

### PR DESCRIPTION
**Pull request by AlexandreSi**

### CODE REVIEW

File: newIDE/app/src/LocalApp.js [hunk 1]
```diff
@@ -48,60 +49,62 @@ export const create = (authentication: Authentication) => {
...
-                gd,
-                objectsEditorService: ObjectsEditorService,
-                objectsRenderingService: ObjectsRenderingService,
-                filterExamples: !isDev,
-              })}
-              initialFileMetadataToOpen={initialFileMetadataToOpen}
-            />
-          )}
-        </ProjectStorageProviders>
+        <ErrorBoundary title="GDevelop encountered an issue">
```
Comment (4ian) [inline newIDE/app/src/LocalApp.js:52]: What do you think of putting this inside `Providers`?

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/LocalApp.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/LocalApp.js]: I'll try!

File: newIDE/app/src/MainFrame/Providers.js [hunk 2]
```diff
@@ -82,52 +83,57 @@ const Providers = ({
...
-                                        </SubscriptionSuggestionProvider>
-                                      </AlertProvider>
-                                    </EventsFunctionsExtensionsProvider>
-                                  )}
-                                </I18n>
-                              </PublicProfileProvider>
-                            </AuthenticatedUserProvider>
-                          </InAppTutorialProvider>
+                          <ErrorBoundary
+                            title="GDevelop encountered an issue"
```
Comment (ClementPasteau) [inline newIDE/app/src/MainFrame/Providers.js:87]: Can this be translated?

Comment (AlexandreSi) [reply newIDE/app/src/MainFrame/Providers.js:87]: I brought the app error boundary above the i18n provider so I would say no 😅

Comment (ClementPasteau) [reply newIDE/app/src/MainFrame/Providers.js:87]: oh yes right

**Pull request merged**